GTT-1487: Notifications process special characters in content item title and dashboard title correctly

yinxudeng opened the pull request:
## Description

- Notifications process special characters in content item title and dashboard title correctly
- Updated bar chart unit tests to resolve warning

## Testing

Passed all unit tests and integration tests. You can also check it out here: https://d1p86h8o9mi768.cloudfront.net/admin/dashboard/edit/14227ccf-e538-49e0-9be1-13bfbb15b11e

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

yinxudeng committed:
fixed edit text widget alert

yinxudeng committed:
fixed edit text widget alert

yinxudeng committed:
special characters now show up correctly when adding and editing content items

yinxudeng committed:
special characters now show up correctly when adding and editing content items

yinxudeng committed:
updated translation file to include prefix and suffix?

yinxudeng committed:
updated translation file to include prefix and suffix?

yinxudeng committed:
display special characters in title

yinxudeng committed:
display special characters in title

yinxudeng committed:
updated barchartwidget test to resolve warning

yinxudeng committed:
ran prettier

yinxudeng committed:
ran prettier

yinxudeng committed:
fixed typo in translation variables

yinxudeng committed:
fixed typo in translation variables

yinxudeng committed:
ran prettier

yinxudeng committed:
ran prettier

trietlu commented on frontend/src/containers/ViewDashboardAdmin.tsx:
```diff
@@ -163,25 +163,33 @@ function ViewDashboardAdmin() {
         isOpen={isOpenRepublishModal}
         closeModal={() => setIsOpenRepublishModal(false)}
-        title={t("RepublishDashboardModalTitle", { name: dashboard?.name })}
+        title={`${t("RepublishDashboardModalTitlePrefix")} ${
```
@yinxudeng instead of "suffix" and "prefix", how about using a hierarchy like
PreparePublishingModalTitle:{
"part1":"something",
"part2":"something else"
}
We do this in other places, and the translator can understand that the texts are related

yinxudeng replied:
Sorry my bad. Using parts makes more sense.

trietlu reviewed (changes requested):
Please see my comment

trietlu committed:
GTT-1498: Change version to 1.0.5.  Instruct user to get code from mainline, and not a particular tag, as unstable changes are now in mainline-dev (#570)

* Change version to 1.0.5.  Instruct user to get code from mainline, and not a particular tag, as unstable changes are now in mainline-dev

* Corrected error in git clone command

Co-authored-by: trietlu <trietlu@amazon.com>

yinxudeng committed:
replaced prefix/suffix with parts in translation file

yinxudeng committed:
replaced prefix/suffix with parts in translation file

yinxudeng committed:
fixed typo

yinxudeng committed:
fixed typo

yinxudeng committed:
updated unit test to resolve warning (#571)

yinxudeng committed:
updated English translation

yinxudeng committed:
updated English translation

trietlu committed:
GTT-1498: Rebase onto mainline-dev from mainline (#574)

* GTT-1403: "Your Logo" should never load once it's replaced (#530)

* #GTT-1436: Fixed bug in auto-top positioning of content item page and used enum to represent percentage and currency (#529)

* GTT-1394: Updated Create Metric Dataset to include symbols and fixed bug with currency in Add Metrics page

* GTT-1394: Localized Currency/Percentage and changed for loop

* GTT-1436: Fixed bug in auto-top positioning of content item page and used enum to represent percentage and currency

* GTT-1436: Added camelCase to variables and passed state setter to custom hook

* GTT-1436: Added camelCase to variables in index.tsx

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

* Have the codepipeline instance be triggered by the mainline-dev branch (#534)

Co-authored-by: trietlu <trietlu@amazon.com>

* Update version to 1.0.4 (#538)

Co-authored-by: trietlu <trietlu@amazon.com>

* #GTT 1455: Replaced exposed emails in postman script with {{$randomEmail}} - mainline (#540)

* Rebase mainline-dev to mainline (#532)

* GTT-1403: "Your Logo" should never load once it's replaced (#530)

* #GTT-1436: Fixed bug in auto-top positioning of content item page and used enum to represent percentage and currency (#529)

* GTT-1394: Updated Create Metric Dataset to include symbols and fixed bug with currency in Add Metrics page

* GTT-1394: Localized Currency/Percentage and changed for loop

* GTT-1436: Fixed bug in auto-top positioning of content item page and used enum to represent percentage and currency

* GTT-1436: Added camelCase to variables and passed state setter to custom hook

* GTT-1436: Added camelCase to variables in index.tsx

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

Co-authored-by: Aakash Balaji <baakash411@gmail.com>
Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

* GTT-1394: Updated Create Metric Dataset to include symbols and fixed bug with currency in Add Metrics page

* GTT-1442: Updated OpenAPI spec, Integration Tests for Create/Update Metrics Dataset and Error Handling/Unit Tests for Update Matrics Dataset

* GTT-1442: Deleted package-lock.json

* GTT-1442: Added enum for percentage and currency

* GTT-1442: Replaced exposed email in postman script with {{}}

Co-authored-by: yinxudeng <47926598+yinxudeng@users.noreply.github.com>
Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

* Bump tar from 6.1.0 to 6.1.5 in /frontend (#559)

Bumps [tar](https://github.com/npm/node-tar) from 6.1.0 to 6.1.5.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v6.1.0...v6.1.5)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump path-parse from 1.0.6 to 1.0.7 in /backend (#565)

Bumps [path-parse](https://github.com/jbgutierrez/path-parse) from 1.0.6 to 1.0.7.
- [Release notes](https://github.com/jbgutierrez/path-parse/releases)
- [Commits](https://github.com/jbgutierrez/path-parse/commits/v1.0.7)

---
updated-dependencies:
- dependency-name: path-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump url-parse from 1.5.1 to 1.5.3 in /frontend (#566)

Bumps [url-parse](https://github.com/unshiftio/url-parse) from 1.5.1 to 1.5.3.
- [Release notes](https://github.com/unshiftio/url-parse/releases)
- [Commits](https://github.com/unshiftio/url-parse/compare/1.5.1...1.5.3)

---
updated-dependencies:
- dependency-name: url-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Triet <triet05@gmail.com>

* GTT-1443: Added testing for changing users' role (#533)

* modified existing users test to account for pagination

* tested changing a user's role in users e2e test

Co-authored-by: Triet <triet05@gmail.com>

* #GTT 1431: Fixed Large Numbers from showing up as infinity and scientific notation for all charts (#544)

* GTT-1394: Updated Create Metric Dataset to include symbols and fixed bug with currency in Add Metrics page

* GTT-1442: Updated OpenAPI spec, Integration Tests for Create/Update Metrics Dataset and Error Handling/Unit Tests for Update Matrics Dataset

* GTT-1442: Deleted package-lock.json

* GTT-1431: Fixed Large Numbers from showing up as infinity and scientif notation for all charts

* GTT-1394: Updated Create Metric Dataset to include symbols and fixed bug with currency in Add Metrics page

* GTT-1442: Deleted package-lock.json

* GTT-1431: Fixed Large Numbers from showing up as infinity and scientif notation for all charts

* GTT-1394: Updated Create Metric Dataset to include symbols and fixed bug with currency in Add Metrics page

* GTT-1442: Deleted package-lock.json

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>

* GTT-1441: Added edit favicon feature to Badger (#536)

* added edit favicon feature

* refactored code

* ran prettier to fix style issue

* fixed code style; added react-helmet license

Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

* GTT-1462: added e2e test for table content item (#547)

* increased content item test coverage; added test for table content item

* refactored code

* removed part of update metrics dataset integration test (#548)

Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

* GTT-1416: Add excel support (#546)

* Add Excel support

* Update package-lock.json

* Refactoring and delete console logs

* Modified cursor to not change when hovering over content item steps (#549)

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

* #GTT-1220: Deleted commented code in cdk folder (#551)

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>

* #GTT-1010: Collected operational metrics PDoA in CFT (#552)

* #GTT-1010: Collected operational metrics PDoA in CFT

* #GTT-1010: added v for version and changed code to per dash code

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>

* Revert "#GTT-1010: Collected operational metrics PDoA in CFT (#552)" (#556)

This reverts commit 5e4874761739ba157206bd50162d0aebecb41383.

* GTT-1419: Added options to preview a dashboard's mobile display (#553)

* added checkbox for mobile preview

* part-to-whole chart shows up on mobile preview, but an error message still lingers in console

* showing Actions on the left in mobile preview

* always display download button and favicon in the same line

* showing Actions on the left in mobile preview for all content items

* show Download button and logo on the left inside mobile preview

* display one metric in each row inside mobile preview

* refactored code; fixed Download position

* updated unit tests

* hide mobile preview checkbox if user is already on mobile device

* refactored code

* #GTT-1488: Updated language on formatting page (#554)

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>

* #GTT-1490: Aligned Edit buttons on Branding and Styling Page (#555)

* #GTT-1488: Updated language on formatting page

* #GTT-1490: Aligned Edit buttons on Branding and Styling Page

* #GTT-1490: Aligned Edit buttons on Branding and Styling Page

* #GTT-1490: Fixed snapshot

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>

* #GTT-1466: Fixed Segments not showing up on Donut/Pie Charts (#550)

Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Triet <triet05@gmail.com>

* GTT-1483: Improved mobile display of pie/donut chart (#557)

* display column chart more spread out on mobile device/small screen (#558)

* ES and PT translations from Totoro localization service (#564)

Co-authored-by: trietlu <trietlu@amazon.com>

* GTT-1516: [Bug] Fixed missing labels in bar chart when there is too much data (#562)

* bar chart height scales dynamically with the number of data

* trigger checks

* externalized logic to calculate chart height and added comments

* refactored code

* updated code logic

Co-authored-by: Triet <triet05@gmail.com>

* [Bug] change grid-col depending on dashboard state (#568)

* GTT-1498: Change version to 1.0.5.  Instruct user to get code from mainline, and not a particular tag, as unstable changes are now in mainline-dev (#570)

* Change version to 1.0.5.  Instruct user to get code from mainline, and not a particular tag, as unstable changes are now in mainline-dev

* Corrected error in git clone command

Co-authored-by: trietlu <trietlu@amazon.com>

* updated unit test to resolve warning (#571)

Co-authored-by: yinxudeng <47926598+yinxudeng@users.noreply.github.com>
Co-authored-by: Aakash Balaji <baakash411@gmail.com>
Co-authored-by: Aakash Balaji <aakbalaj@amazon.com>
Co-authored-by: Miguel Pavon Diaz <71112226+miguelpdiaz8@users.noreply.github.com>
Co-authored-by: trietlu <trietlu@amazon.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

yinxudeng committed:
finalized English translation

yinxudeng committed:
finalized English translation

yinxudeng committed:
updated Portuguese translation

yinxudeng committed:
updated Portuguese translation

yinxudeng committed:
updated Spanish translation:

yinxudeng committed:
updated Spanish translation:

yinxudeng committed:
Merge branch 'GTT-1487' of https://github.com/awslabs/performance-dashboard-on-aws into GTT-1487

yinxudeng committed:
minor fix to spanish translation
